Geometry/CSCGeometryBuilder/src/CSCGeometryParsFromDD.cc

```diff
@@ -301,3 +358,272 @@ bool CSCGeometryParsFromDD::build(const DDCompactView* cview,
+      }
+
+      auto wirespacing = fv.get<double>("WireSpacing");
+      wg.wireSpacing = static_cast<double>(wirespacing);
+
+      auto alignmentpintofirstwire = fv.get<double>("AlignmentPinToFirstWire");
+      wg.alignmentPinToFirstWire = static_cast<double>(alignmentpintofirstwire);
+
+      auto totnumwiregroups = fv.get<double>("TotNumWireGroups");
+      wg.numberOfGroups = static_cast<int>(totnumwiregroups);
+
+      auto lengthoffirstwire = fv.get<double>("LengthOfFirstWire");
+      wg.narrowWidthOfWirePlane = static_cast<double>(lengthoffirstwire);
+
+      auto lengthoflastwire = fv.get<double>("LengthOfLastWire");
+      wg.wideWidthOfWirePlane = static_cast<double>(lengthoflastwire);
+
+      auto radialextentofwireplane = fv.get<double>("RadialExtentOfWirePlane");
+      wg.lengthOfWirePlane = static_cast<double>(radialextentofwireplane);
+
+      uparvals.emplace_back((wg.wireSpacing) * 10.0);
+      uparvals.emplace_back(wg.alignmentPinToFirstWire);
+      uparvals.emplace_back(wg.numberOfGroups);
+      uparvals.emplace_back(wg.narrowWidthOfWirePlane);
+      uparvals.emplace_back(wg.wideWidthOfWirePlane);
+      uparvals.emplace_back(wg.lengthOfWirePlane);
+      uparvals.emplace_back(wg.wiresInEachGroup.size());
+
+      for (CSCWireGroupPackage::Container::const_iterator it = wg.wiresInEachGroup.begin();
+           it != wg.wiresInEachGroup.end();
+           ++it) {
+        uparvals.emplace_back(*it);
+      }
+      for (CSCWireGroupPackage::Container::const_iterator it = wg.consecutiveGroups.begin();
+           it != wg.consecutiveGroups.end();
+           ++it) {
+        uparvals.emplace_back(*it);
+      }
+
+      /** end stuff **/
+    }
+
+    fpar.clear();
+
+    std::string my_title(fv.solid()->GetTitle());
+
+    if (my_title == "Subtraction") {
+      cms::DDSolid mysolid(fv.solid());
+      auto solidA = mysolid.solidA();
+      std::vector<double> dpar = solidA.dimensions();
+
+      std::transform(
+          dpar.begin(), dpar.end(), dpar.begin(), [](double i) -> double { return cms_rounding::roundIfNear0(i); });
+
+      fpar.emplace_back((dpar[1]));
+      fpar.emplace_back((dpar[2]));
+      fpar.emplace_back((dpar[3]));
+      fpar.emplace_back((dpar[4]));
+    } else {
+      dpar = fv.parameters();
+
+      std::transform(
+          dpar.begin(), dpar.end(), dpar.begin(), [](double i) -> double { return cms_rounding::roundIfNear0(i); });
+
+      fpar.emplace_back((dpar[0]));
+      fpar.emplace_back((dpar[1]));
+      fpar.emplace_back((dpar[2]));
+      fpar.emplace_back((dpar[3]));
+    }
+
+    gtran[0] = (float)1.0 * (fv.translation().X());
+    gtran[1] = (float)1.0 * (fv.translation().Y());
+    gtran[2] = (float)1.0 * (fv.translation().Z());
+
+    if ((-1.0e-10 < gtran[0]) && (gtran[0] < 1.0e-10))
+      gtran[0] = 0.0;
+    if ((-1.0e-10 < gtran[1]) && (gtran[1] < 1.0e-10))
+      gtran[1] = 0.0;
+    if ((-1.0e-10 < gtran[2]) && (gtran[2] < 1.0e-10))
+      gtran[2] = 0.0;
```

Comment:
You're already using roundIfNear0(). Use it on lines 514-519 with `1.e-10` as the second parameter.

Reply:
Done